Assets/CesiumForUnitySamples/Scripts/CesiumSamplesRequiresMagicLeap.cs

```diff
@@ -0,0 +1,243 @@
+            },
+            () =>
+            {
+                OpenXRSettings oxrSettings = OpenXRSettings.GetSettingsForBuildTargetGroup(BuildTargetGroup.Android);
+
+                OpenXRFeature[] interactionFeatures = oxrSettings.GetFeatures<OpenXRInteractionFeature>();
+                MagicLeapControllerProfile magicLeapProfile = interactionFeatures.FirstOrDefault(f => f.GetType() == typeof(MagicLeapControllerProfile)) as MagicLeapControllerProfile;
+                if (magicLeapProfile != null)
+                {
+                    magicLeapProfile.enabled = true;
+                }
+            }),
+        // We need to disable ML's handling of the far clip plane, so we can see distant terrain
+        // This property is private so we use SerializedProperty to access it
+        new Case(
+            () =>
+            {
+                OpenXRSettings oxrSettings = OpenXRSettings.GetSettingsForBuildTargetGroup(BuildTargetGroup.Android);
+                MagicLeapFeature mlFeature = oxrSettings.GetFeature<MagicLeapFeature>();
+                return mlFeature != null && mlFeature.enabled && mlFeature.FarClipPolicy == MagicLeapFeature.FarClipMode.None;
+            },
+            () =>
+            {
+                OpenXRSettings oxrSettings = OpenXRSettings.GetSettingsForBuildTargetGroup(BuildTargetGroup.Android);
+                MagicLeapFeature mlFeature = oxrSettings.GetFeature<MagicLeapFeature>();
+                mlFeature.enabled = true;
+
+                SerializedObject mlFeatureObject = new SerializedObject(mlFeature);
+                SerializedProperty farClipProperty = mlFeatureObject.FindProperty("farClipPolicy");
+                farClipProperty.intValue = (int)MagicLeapFeature.FarClipMode.None;
+                mlFeatureObject.ApplyModifiedPropertiesWithoutUndo();
+            })
+#endif // CESIUM_MAGIC_LEAP
+    };
+
+    static CesiumSamplesRequiresMagicLeap()
+    {
+        EditorApplication.playModeStateChanged += OnStateChanged;
+    }
+#endif // UNITY_2022_2_OR_NEWER
+
+    private void Start()
+    {
+        // There should only ever be one of these at most in a scene.
+        if (_instance != null && _instance != this)
+        {
+            DestroyImmediate(this);
+            return;
+        }
+
+        _instance = this;
+
+#if !UNITY_2022_2_OR_NEWER
+        EditorUtility.DisplayDialog("Unity 2022.2 or Greater Required", ML_MIN_VERSION_TEXT, "Ok");
+        return;
+#endif // !UNITY_2022_2_OR_NEWER
+
+#if UNITY_2022_2_OR_NEWER
+        if (_waitingForReturnToEditMode)
+        {
+            return;
+        }
+
+        bool hasMagicLeap = UnityEditor.PackageManager.PackageInfo.GetAllRegisteredPackages().Any(p => p.name == "com.magicleap.unitysdk");
+        if (!hasMagicLeap)
+        {
+            if (EditorUtility.DisplayDialog("Setup required", ML_PACKAGE_REQUIRED_TEXT, "Open documentation", "I'll add it manually"))
```

Comment:
If I may nitpick these: 

```suggestion
            if (EditorUtility.DisplayDialog("Magic Leap SDK Setup required", ML_PACKAGE_REQUIRED_TEXT, "Open Magic Leap documentation", "Ok"))
```